src/utils/Multicall.ts

```diff
@@ -94,3 +96,32 @@ export async function blockAndAggregate(
 
   return { blockNumber: blockNumber.toNumber(), returnData };
 }
+
+async function isDeployed(provider: providers.Provider): Promise<boolean> {
+  const chainId = (await provider.getNetwork()).chainId;
+  const expectedMulticallAddress = getMulticallAddress(chainId) ?? DETERMINISTIC_MULTICALL_ADDRESS;
+  const code = await provider.getCode(expectedMulticallAddress);
+  return code !== "0x";
+}
+
+/**
+ * Deploy a Multicall3 instance.
+ * Guidance: https://github.com/mds1/multicall3?tab=readme-ov-file#new-deployments
+ * @param signer An Ethers Signer instance with at least 0.1 ETH.
+ */
+export async function deploy(signer: Signer) {
+  assert(signer.provider);
```

Comment:
@bmzig OK to restrict this when chainId is a testnet? The Multicall3 docs have some pretty strict wording about not messing up the deployment because you only get one shot at it, so it'd be nice for me to not be the reason that a chain can't have a nice deterministic address :sweat_smile:

Reply:
I think this does it, right? https://github.com/across-protocol/sdk/pull/920/commits/2c9fefee8d4ed67ebdc35fb00ed997b69c76ecb2 (also made some mistakes in that commit that I has to fix here https://github.com/across-protocol/sdk/pull/920/commits/f669d1e675c782ab3114a06c503d3e0f4a740aad)